src/storage/src/upsert_continual_feedback.rs

```diff
@@ -246,170 +275,115 @@ where
                                     ?upper,
                                     "received persist progress");
                                 persist_upper = upper;
+
+                                ingesting_persist_timestamps.retain(|ts| persist_upper.less_equal(ts));
                             }
                         }
                     }
```

Comment:
It's explained by this comment on the variable: https://github.com/MaterializeInc/materialize/blob/b104dde49ff6b46e908d6b57ad6750cf96f51a47/src/storage/src/upsert_continual_feedback.rs#L242

As soon as we ingest some data for a timestamp, the `Consolidating` entry in upsert state might be inconsistent (you haven't seen all the diffs yet, so they don't sum up to zero or on). If you then tried to decode it, it would fail. Only when you've seen all data for that timestamp, i.e. when it's not beyond the frontier anymore, can you proceed with accessing state.

> but that's not necessarily the case since the timely timestamp is less than or equal to the ts of the DD update

The rehydration code in the existing UPSERT operator uses the timely frontier to determine when it's done, wouldn't that also be problematic when the timestamps in the data are beyond the timely timestamp? I can change this code to instead sniff out the timestamps from the data, but that would be much more expensive.